### PR TITLE
Fix Analizer Rules (Non Cms Modules)

### PR DESCRIPTION
**Pull request by jtkech**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Workflows/Http/Drivers/HttpRequestFilterEventDisplayDriver.cs [hunk 1]
```diff
@@ -22,9 +22,9 @@ protected override void UpdateActivity(HttpRequestFilterEventViewModel model, Ht
             activity.AreaName = NullIfEmpty(model.AreaName);
         }
 
-        private string NullIfEmpty(string s)
+        private static string NullIfEmpty(string s)
         {
-            return string.IsNullOrWhiteSpace(s) ? null : s.Trim();
+            return System.String.IsNullOrWhiteSpace(s) ? null : s.Trim();
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Workflows/Http/Drivers/HttpRequestFilterEventDisplayDriver.cs:27]: Using System;

**Comment by MikeAlhayek** [comment]
Not a fan of the #pragma blocks and previously discuss. I suggest we add `_s` and `_h` in the language extractor and rename the variables

File: src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ExportRemoteInstanceController.cs [hunk 2]
```diff
@@ -20,14 +21,16 @@ namespace OrchardCore.Deployment.Remote.Controllers
     [Admin]
     public class ExportRemoteInstanceController : Controller
     {
-        private static readonly HttpClient _httpClient = new HttpClient();
+        private static readonly HttpClient _httpClient = new();
```
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/ExportRemoteInstanceController.cs:24]: I don't know, could be suggested in another PR, here we define a static reusable HttpClient, maybe IHttpClientFactory was not existing when it has been done ;)

File: src/OrchardCore.Modules/OrchardCore.Resources/Liquid/LinkTag.cs [hunk 3]
```diff
@@ -42,27 +44,27 @@ public static async ValueTask<Completion> WriteToAsync(List<FilterArgument> argu
 
             var linkEntry = new LinkEntry();
 
-            if (!string.IsNullOrEmpty(src))
+            if (!System.String.IsNullOrEmpty(src))
```
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Resources/Liquid/LinkTag.cs:47]: Yes, will do

File: src/OrchardCore.Modules/OrchardCore.Twitter/Services/TwitterClientMessageHandler.cs [hunk 4]
```diff
@@ -93,15 +94,16 @@ public async Task ConfigureOAuthAsync(HttpRequestMessage request)
...
                 Uri.EscapeDataString(request.RequestUri.AbsoluteUri.ToString()), "&",
-                Uri.EscapeDataString(string.Join("&", sortedParameters.Select(c => string.Format("{0}={1}", c.Key, c.Value)))));
+                Uri.EscapeDataString(String.Join("&", sortedParameters.Select(c => String.Format("{0}={1}", c.Key, c.Value)))));
+
+            var secret = String.Concat(_twitterSettings.ConsumerSecret, "&", _twitterSettings.AccessTokenSecret);
 
-            var secret = string.Concat(_twitterSettings.ConsumerSecret, "&", _twitterSettings.AccessTokenSecret);
             string signature;
-            using (var hasher = new HMACSHA1(ASCIIEncoding.ASCII.GetBytes(secret)))
+            using (var hasher = new HMACSHA1(Encoding.ASCII.GetBytes(secret)))
```
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Twitter/Services/TwitterClientMessageHandler.cs:104]: Here we can't as it doesn't apply to whole parent block

File: src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/RegisterUserTask.cs [hunk 5]
```diff
@@ -148,10 +151,7 @@ public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecuti
                     if (smtpService == null)
                     {
                         var updater = _updateModelAccessor.ModelUpdater;
-                        if (updater != null)
-                        {
-                            updater.ModelState.TryAddModelError("", S["No email service is available"]);
-                        }
+                        updater?.ModelState.TryAddModelError("", S["No email service is available"]);
```
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Users/Workflows/Activities/RegisterUserTask.cs:154]: Yes I prefer too but as said it is not part of our current rules and there are many ones, so maybe in a subsequent PR.

File: src/OrchardCore.Modules/OrchardCore.Workflows/Http/Activities/HttpRequestTask.cs [hunk 6]
```diff
@@ -83,9 +83,11 @@ public class HttpRequestTask : TaskActivity
             { 599 , "Network Connect Timeout Error" }
         };
 
-        private static readonly HttpClient _httpClient = new HttpClient();
+        private static readonly HttpClient _httpClient = new();
```
[the remark this answers is not in the document]
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Http/Activities/HttpRequestTask.cs:86]: Maybe ;)

File: src/OrchardCore.Modules/OrchardCore.Workflows/Http/Drivers/HttpRequestFilterEventDisplayDriver.cs (hunk 1 above)
Comment (jtkech) [reply src/OrchardCore.Modules/OrchardCore.Workflows/Http/Drivers/HttpRequestFilterEventDisplayDriver.cs:27]: Yes, will do

**Comment by jtkech** [comment]
> Not a fan of the #pragma blocks and previously discuss. I suggest we add _s and _h in the language extractor and rename the variables

Yes I agree, we should plan to do this

**Comment by MikeAlhayek** [comment]
> Yes I agree, we should plan to do this

I suggest we do it before merging this PR or make it part of this PR. If we use `pragma` block and try to change it later, we may not be able to do it easily with `dotnet format`

**Comment by jtkech** [comment]
Easy to do, we will just have to search for all occurences of `S;` and `H;` and remove the related pragmas.

**Comment by MikeAlhayek** [comment]
Yes but much easier if we don't have the pragma and just let the formatter rename things. I am wondering what is the effort needed to modify the extractor? Never visited that code for it. I always wondered why don't we have reflection based extractor too instead... but maybe a discussion for a different day :)

**Comment by jtkech** [comment]
Never worked on this tool.

Hmm, maybe I have an idea, why not defining all these fields as `protected` in place of `private` so that they go outside the rule, I think some of them are already declared like this.

I can do this update for all my Analyzer Rules PRs

**Comment by MikeAlhayek** [comment]
Yes declaring as protected is definitely better than using pragma block

File: src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/RemoteInstanceController.cs [hunk 7]
```diff
@@ -60,12 +61,12 @@ public async Task<IActionResult> Index(ContentOptions options, PagerParameters p
...
             var remoteInstances = (await _service.GetRemoteInstanceListAsync()).RemoteInstances;
 
-            if (!string.IsNullOrWhiteSpace(options.Search))
+            if (!String.IsNullOrWhiteSpace(options.Search))
             {
                 remoteInstances = remoteInstances.Where(x => x.Name.Contains(options.Search, StringComparison.OrdinalIgnoreCase)).ToList();
             }
 
-            var count = remoteInstances.Count();
+            var count = remoteInstances.Count;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Deployment.Remote/Controllers/RemoteInstanceController.cs:69]: This line can be removed and the call to .Count can be moved down

File: src/OrchardCore.Modules/OrchardCore.Features/Controllers/AdminController.cs [hunk 8]
```diff
@@ -23,33 +23,34 @@ public class AdminController : Controller
...
             IAuthorizationService authorizationService,
             IShellHost shellHost,
             ShellSettings shellSettings,
+            IExtensionManager extensionManager,
+            IShellFeaturesManager shellFeaturesManager,
+            IOptions<AdminOptions> adminOptions,
             INotifier notifier,
             IStringLocalizer<AdminController> stringLocalizer,
-            IOptions<AdminOptions> adminOptions)
+            IHtmlLocalizer<AdminController> localizer)
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Features/Controllers/AdminController.cs:43]: htmlLocalizer

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**